server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java

```diff
@@ -522,6 +524,17 @@ protected static DirectoryReader openDirectory(Directory directory, boolean wrap
         }
     }
 
+    protected static DirectoryReader openDirectory(Directory directory, boolean wrapSoftDeletes, Comparator<LeafReader> leafSorter)
```

Comment:
Instead of overloading this method, can we safely update/replace the version directly above this? 

It looks like the previous method was only used by `NoOpEngine`, and you've updated that call to pass `leafSorter`. So, I don't think we need the version that doesn't take `leafSorter` anymore. Since `ReadOnlyEngine` is not tagged as `@PublicApi`, I think it should be safe to update the existing method without breaking backward-compatibility.